app/resource.py

```diff
@@ -80,13 +80,16 @@ def is_safe_to_stop(self, today_date=util.TODAY_YYYY_MM_DD):
 
     # Default implementation. Individual resources defines the implementation.
     def can_be_terminated(self, today_date=util.TODAY_YYYY_MM_DD):
-        return False
+        parsed_date: parsing.ParsedDate = parsing.parse_date_tag(self.terminate_after)
+        return not (len(self.eks_nodegroup_name) > 0) and util.has_date_passed(parsed_date.expiry_date, today_date)
```

Comment:
Does this make sense to be here if only instances will have `eks_nodegroup_name`?

Reply:
Yes, in that case, it does make sense to keep it for instances only. I should have explored more to know which resources can have `eks_nodegroup_name`